homeassistant/components/airzone/climate.py

```diff
@@ -184,4 +181,7 @@ def _async_update_attrs(self) -> None:
         else:
             self._attr_hvac_action = CURRENT_HVAC_OFF
             self._attr_hvac_mode = HVAC_MODE_OFF
+        self._attr_hvac_modes = [
+            HVAC_MODE_LIB_TO_HASS[mode] for mode in self.get_zone_value(AZD_MODES)
+        ]
```

Comment:
We don't allow `_attr_hvac_modes` to change after the start event since it will break anything that looks at the value (ie Voice Assistants such as Aleax, Google Assistant, HomeKit, etc)

Reply:
Instead set all the supported modes the device can support and raise HomeAssistantError if they choose a mode that isn't available  in the current state

Reply:
Ok, I will do it that way then

Reply:
@bdraco introduced the suggested changes on 692bd223f02979ef17d57f01704bbad60f1f44ee

Reply:
And I forgot to revert the HVAC modes change, but it should be solved in c1753c0a859210581ff1823acadb50955258a655.
Sorry for that!